File: ios/My Job Pitch/My Job Pitch/API/Job.m
//
//  Job.m
//  My Job Pitch
//
//  Created by user on 12/01/2016.
//  Copyright © 2016 SC Labs Ltd. All rights reserved.
//

#import "Job.h"

@implementation Job

- (Image*)getImage
{
    if (self.images && self.images.count > 0) {
        return [self.images firstObject];
    } else {
        return [self.locationData getImage];
    }
}

@end

File: ios/My Job Pitch-Swift/MyJobPitch/Models/api/Job.swift
//
//  Job.swift
//  MyJobPitch
//
//  Created by dev on 12/20/16.
//  Copyright © 2016 myjobpitch. All rights reserved.
//

import Foundation

class Job: MJPObjectWithDates {

    var title: String!
    var desc: String!
    var sector: NSNumber!
    var location: NSNumber!
    var contract: NSNumber!
    var hours: NSNumber!
    var status: NSNumber!
    var locationData: Location!
    var images: NSArray!
    var videos: NSArray!
    
    func getPitch() -> JobPitch? {
        if videos != nil {
            for pitch in videos as! [JobPitch] {
                if pitch.video != nil {
                    return pitch
                }
            }
        }
        return nil
    }
 
    func getImage() -> Image? {
        if images != nil && images.count > 0 {
            return images.firstObject as? Image
        }
        return locationData.getImage()
    }
    
    func getBusinessName() -> String {
        return locationData.businessData.name + ", " + locationData.name
    }
}